### PR TITLE
Closing in on the qgisplugin gap

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
- [x] moved `Path` to internals, using `str` in user-facing functions
- [x] less printing, more logging
- [x] less positional named parameters, more named parameters (**this breaks compatibility with qgis-plugin-ci and other consumers; adjustments asked from consumer code: see comments below in this PR**)
- [x] 100% function-parity with older `api.py` module
- [x] added test for public repositories
- [x] passing test for public repositoies
- [ ] fix `list_languages` (waiting on https://community.transifex.com/t/how-to-get-a-list-of-languages-to-which-a-specific-resource-was-translated/3300)

### CODE REVIEW

File: pytransifex/api.py [hunk 1]
```diff
@@ -69,6 +68,7 @@ def create_project(
                     source_language=source_language,
                     private=private,
                     organization=self.organization,
+                    **kwargs,
```
Comment (why-not-try-calmer) [inline pytransifex/api.py]: **Potential compatibility issue**: This function now disallows positional arguments.

File: pytransifex/api.py [hunk 2]
```diff
@@ -144,20 +148,20 @@ def create_resource(
...
         else:
-            raise Exception(
+            raise ValueError(
                 f"Not project could be found wiht the slug '{project_slug}'. Please create a project first."
             )
 
     @ensure_login
     def update_source_translation(
-        self, project_slug: str, resource_slug: str, path_to_file: Path
+        self, *, project_slug: str, resource_slug: str, path_to_file: str
```
Comment (why-not-try-calmer) [inline pytransifex/api.py:157]: **Potential compatibility issue**: This function now disallows posittional arguments.

File: pytransifex/api.py [hunk 3]
```diff
@@ -175,21 +179,23 @@ def update_source_translation(
...
     @ensure_login
     def get_translation(
         self,
+        *,
         project_slug: str,
         resource_slug: str,
         language_code: str,
-        output_dir: Path,
-    ):
+        path_to_output_dir: str,
```
Comment (why-not-try-calmer) [inline pytransifex/api.py]: **Potential compatibility issue**: `output_dir` renamed to `path_to_output_dir` for the sake of expliciteness.

File: pytransifex/api.py [hunk 4]
```diff
@@ -198,51 +204,55 @@ def get_translation(
...
                 )
         else:
-            raise Exception(
+            raise ValueError(
                 f"Couldn't find any project with this slug: '{project_slug}'"
             )
 
     @ensure_login
-    def list_languages(self, project_slug: str) -> list[Any]:
+    def list_languages(self, project_slug: str, resource_slug: str) -> list[Any]:
```
Comment (why-not-try-calmer) [inline pytransifex/api.py]: This function was not implemented correctly. This fixes that.

File: pytransifex/api.py [hunk 5]
```diff
@@ -304,12 +323,12 @@ def pull(
 
     @ensure_login
     def push(
-        self, project_slug: str, resource_slugs: list[str], path_to_files: list[Path]
+        self, *, project_slug: str, resource_slugs: list[str], path_to_files: list[str]
```
Comment (why-not-try-calmer) [inline pytransifex/api.py:332]: **Potential compatibility issue**: This function now disallows posittional arguments.

File: pytransifex/api.py [hunk 6]
```diff
@@ -185,11 +189,24 @@ def get_translation(
         project_slug: str,
         resource_slug: str,
         language_code: str,
-        output_dir: Path,
-    ):
+        path_to_output_file: None | str = None,
```
Comment (why-not-try-calmer) [inline pytransifex/api.py:176]: **Renamed named parameter**: breaks compatibility!

File: pytransifex/api.py [hunk 7]
```diff
@@ -198,54 +199,69 @@ def get_translation(
...
-            raise Exception(
+                if resource := project.fetch("resources").get(slug=resource_slug):
+                    it = tx_api.ResourceLanguageStats.filter(
+                        project=project, resource=resource
+                    ).all()
+
+                    language_codes = []
+                    for tr in it:
+                        """
+                        FIXME
```
Comment (why-not-try-calmer) [inline pytransifex/api.py:238]: FIX ME

**Review by 3nids — APPROVED** [review]

**Pull request merged**